src/lib.rs

```diff
@@ -108,10 +108,18 @@ fn path_to_cstring(path: impl AsRef<Path>) -> Result<CString> {
     Ok(CString::new(s)?)
 }
 
+fn to_i32(value: usize, task: ErrorTask) -> Result<i32> {
+    value.try_into().map_err(|e| Error::NumericCastFailed {
+        source: e,
+        value,
+        task,
+    })
+}
```

Comment:
With an extra method to cast into the correct type, i think it makes more sense to cast directly into `std::os::raw::c_int` instead of `i32`. This will make the code also work on systems that use i16 or i64 for c_int.

Reply:
Done! I also changed a few other local variables from `i32`/`f32` to `c_int`/`c_float`, but only when it comes from within a function.